sdk/instrumentation/opentelemetry-instrumentation-azure-sdk/test/public/util/testClient.ts

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { OperationTracingOptions, TracingClient, createTracingClient } from "@azure/core-tracing";
+import {
+  Pipeline,
+  PipelineResponse,
+  createDefaultHttpClient,
+  createHttpHeaders,
+  createPipelineFromOptions,
+} from "@azure/core-rest-pipeline";
+
+import { SDK_VERSION } from "../../../src/constants";
+
+/**
+ * A partial interface compatible with OperationOptions.
+ */
+export interface Options {
+  tracingOptions?: OperationTracingOptions;
+  throw?: boolean;
+}
+
+/**
+ * Options passed to {@link createTracingClient}, exported for validation.
+ */
+export const tracingClientAttributes = {
+  namespace: "Microsoft.Test",
+  packageName: "@azure/opentelemetry-instrumentation-azure-sdk",
+  packageVersion: SDK_VERSION,
+};
+
+/**
+ * A small client that uses core-rest-pipeline to make requests.
+ */
+export class TestClient {
+  tracingClient: TracingClient;
+  pipeline: Pipeline;
+
+  constructor() {
+    this.tracingClient = createTracingClient(tracingClientAttributes);
+    this.pipeline = createPipelineFromOptions({});
+  }
+
+  /**
+   * The entrypoint of this client, which the tests will call into
+   */
+  async outer(options: Options = {}): Promise<PipelineResponse> {
```

Comment:
nit: this function name confused me a little bit when reading the test first. I thought it was some kind of obscure JS feature. Looking at the class it is clear. I wonder if there is a friendlier name we can use

Reply:
Perhaps something like `exampleOperation` since it's mimicking a public client operation? Then `inner` could be `generatedClientOperation` or some such?